### ISSUE TITLE
Black screen after trying the fix

### ISSUE THREAD

**Issue by Mikaka27** [fakhraldin/mesa-git-radv-patched-25659#1]
Don't what to add here, besides logs from lutris:
```
lutris-wrapper: Alan Wake 2
Started initial process 40751 from gamemoderun no-internet /home/mikaka/.local/share/lutris/runners/wine/wine-ge-8-21-x86_64/bin/wine /mnt/E/Gry/Alan Wake 2/AlanWake2.exe
Start monitoring process.
gamemodeauto: 
ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
fsync: up and running.
wine: Using setpriority to control niceness in the [-10,10] range
Initial process has exited (return code: 0)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by ntlm_auth)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libgensec-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libgse-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/libsmbconf.so.0)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/libsamba-util.so.0)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libauthkrb5-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libdbwrap-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libsamba-sockets-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libcli-cldap-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libaddns-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libcli-nbt-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libsamdb-common-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/libldb.so.2)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libmessages-dgm-samba4.so)
ntlm_auth: /usr/lib/libtevent.so.0: version `TEVENT_0.15.0' not found (required by /usr/lib/samba/libclidns-samba4.so)
```

Yea, right system information would be good, it's manjaro on kde plasma (stable branch):
```
Operating System: Manjaro Linux 
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.11
Kernel Version: 6.5.5-1-MANJARO (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® Core™ i5-6600K CPU @ 3.50GHz
Memory: 23.4 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 530
Manufacturer: MSI
Product Name: MS-7971
System Version: 1.0
```

It says that I have Intel HD Graphics 530, that's because my monitor is connected to motherboard, I also have RX 6700 XT, for gaming.

**Comment by TheAnnoyingFruit** [comment]
I also experience this on arch with gnome

**Comment by fakhraldin** [comment]
@Mikaka27  Thanks for the logs. 
I am missing the typical message "Monitored process exited.", when the app gets terminated. This fits the issue description of a black screen resulting from no signal output.
Other than that i can't spot anything unusual. The ntlm_auth messages seems to stem from an app inside of your wine-prefix but not from AW2. The latter doesn't usually make calls for samba at least not in my log.

> It says that I have Intel HD Graphics 530, that's because my monitor is connected to motherboard, I also have RX 6700 XT, for gaming.

The second part of your info about a multi gpu setup though might give us a hint. It might be worth a try to detach your monitor cable from your motherboard, attach it directly to you RX 6700 XT and then to repeat the test. Please shut off your computer first, before doing that and reboot.

**Comment by fakhraldin** [comment]
@TheAnnoyingFruit  
> I also experience this on arch with gnome

~Do you also have a multi gpu setup? Can you please share a log and some system infos?~

It seems my package is incompatible with arch. Luckily @Mikaka27 managed to compile a patched driver for arch.

**Comment by Mikaka27** [comment]
I don't think this is it, I've compiled the mesa with the patches you specified myself, and it works fine. I could share it, however it is much bigger than what you released, I would have to look what I can exclude from the compiled files.

**Comment by fakhraldin** [comment]
> I've compiled the mesa with the patches you specified myself, and it works fine

Nice! 

>  I could share it, however it is much bigger than what you released, I would have to look what I can exclude from the compiled files.

You could actually upload and publish the driver as a zip release, like i did. That way you can share files bigger than 25 MB in gitub. However that's up to you.

> I could share it, however it is much bigger than what you released, I would have to look what I can exclude from the compiled files.

Yes i see that your driver built is 1.14 GB big. It seems you have many things in there, which you don't actually need. There is a way to reduce that. Have a look at my follow up tutorial starting here.:
https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1786192456

**Comment by TheAnnoyingFruit** [comment]
Yes if you could upload it @Mikaka27 that would be awesome

**Comment by Mikaka27** [comment]
> Yes if you could upload it @Mikaka27 that would be awesome

Here you go, I hope it will work for you:
https://github.com/Mikaka27/mesa-git-radv-patched-25659

My release is much bigger, I'm not sure why, perhaps @fakhraldin used different compilation options.

**Comment by fakhraldin** [comment]
@TheAnnoyingFruit Don't worry.
I think i am going to write a small and quick tutorial for you on how you can compile the patched driver by yourself. Sadly i can not test this on my distro. So you have to try this on your machine and give feedback, how it works. Ok?

EDIT: Ok i now saw that @Mikaka27   already uploaded a patched driver. Nice!

**Comment by TheAnnoyingFruit** [comment]
Perhaps that would be helpful still because I have tried @Mikaka27 's driver and it still has a black screen

**Comment by fakhraldin** [comment]
First of all you need the following dependencies as a preparation for the compilation. On fedora i do this by entering the command below. The culprit is though you might run into missing packages on arch and i don't know the equivalent package names on arch.

`dnf install git meson ninja-build gcc gcc-c++ cmake.i686 cmake.x86_64 glibc-devel.i686 glibc-devel.x86_64 valgrind-devel.i686 valgrind-devel.x86_64 bison flex vulkan*.x86_64 vulkan*.i686 libdrm-devel.x86_64 libdrm-devel.i686 libvdpau-devel.x86_64 libvdpau-devel.i686 libva-devel.x86_64 libva-devel.i686 libomxil-bellagio-devel.x86_64 libomxil-bellagio-devel.i686 zlib-devel.x86_64 zlib-devel.i686 llvm-devel.x86_64 llvm-devel.i686 elfutils-libelf-devel.x86_64 elfutils-libelf-devel.i686 wayland*-devel.x86_64 wayland*-devel.i686 wayland-protocols-devel libX*-devel.x86_64 libX*-devel.i686 libunwind-devel.x86_64 libunwind-devel.i686 libxshmfence-devel.x86_64 libxshmfence-devel.i686 lm_sensors-devel.x86_64 lm_sensors-devel.i686 expat-devel.x86_64 expat-devel.i686 libzstd-devel.x86_64 libzstd-devel.i686 pkgconf-pkg-config.i686 pkgconf-pkg-config.x86_64 libffi-devel.i686 libffi-devel.x86_64 libxcb-devel.i686 libxcb-devel.x86_64`

Note: Being on arch you have to replace `dnf install` with `pacman -S`

**Comment by TheAnnoyingFruit** [comment]
ok. I have acquired all the dependencies I could that had a similarly named arch package

**Comment by fakhraldin** [comment]
Very good. If you have installed all dependencies above, then do these next steps.:

1.)  Open a terminal and create a directory called "Projects" in your user profile /home/(YOUR-USER-PROFILE-NAME)/
Don't forget to replace (YOUR-USER-PROFILE-NAME)  with your actual user name!
`mkdir -p ~/Projects`
Enter this Projects directory
`cd ~/Projects`
Download a copy of mesa-git by entering this command
`git clone --depth=1 https://gitlab.freedesktop.org/mesa/mesa.git`
you should now have a new directory in Projects called mesa: /home/(YOUR-USER-PROFILE-NAME)/Projects/mesa

2.) Open your browser and go to the mesa-git project website. Download the needed patches each.
Press the blue code button in the right top corner of each three Merge Requests and select download "plain diff"
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25530
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25659
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25890
You should now have all three diff files in your home's Downloads directory: 
`25530.diff, 25659.diff and 25890.diff`
Copy all of them into our newly created Directory:  /home/YOUR-USER-PROFILE-NAME)/Projects/mesa
You can use your file manager as an easier means.
Then go back to your open terminal. Enter the mesa directory /home/YOUR-USER-PROFILE-NAME)/Projects/mesa
`cd mesa`
From inside of this mesa directory enter the following commands step by step and in chronological order.
We are going to use the --verbose parameter to get visual feedback if the patching succeeded or not. 
You should get a success message after each entered command! Otherwise something went wrong.
`git apply 25530.diff --verbose`
`git apply 25659.diff --verbose`
`git apply 25890.diff --verbose`

to be continued. I am writing the second part right now.

UPDATE 2023 Nov 02: Caution! Patching with MR 25659 is no longer needed, as it has been merged into mesa-git about 20 minutes ago. https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1790725860

**Comment by TheAnnoyingFruit** [comment]
I assume this line git apply 21929.diff --verbose should be 25530

**Comment by fakhraldin** [comment]
3.)  As a next stept we are going to configure the build with meson. Enter this command:
`meson build64 --libdir lib64 --prefix $HOME/mesa -Dgallium-drivers=radeonsi,swrast,zink -Dvulkan-drivers=amd -Dgallium-nine=true -Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec -Dbuildtype=release`

4.) Now we are going to compile with ninja. This can take a couple of minutes depending on how fast your machine is. 
`ninja -C build64 install`

5.) After that you should see a new directory:  /home/YOUR-USER-PROFILE-NAME)/mesa
Verify this directory by entering this command:
`cd ~/mesa && ls -la`
You should see three directories there called `include`, `lib64` and `share`
That's it! Leave that mesa directory as it is and follow the instructions provided by the main project of this repo.
Jump to chapter "3.) Procedure" and follow the further steps from Point 4 on until the end.

**Comment by fakhraldin** [comment]
> I assume this line git apply 21929.diff --verbose should be 25530

Yes my bad it should be `git apply 25530.diff --verbose`
Corrected, Thanks!

**Comment by TheAnnoyingFruit** [comment]
NOW WORKING. thank you so much

**Comment by fakhraldin** [comment]
Glad to help :) This makes me very happy!

**Comment by Tnnk3** [comment]
Hi, i get same black screen, having trouble following compilation some commands is not working, im on steamos steamdeck. Thanks!

**Comment by fakhraldin** [comment]
SteamOS is an arch based distro and should actually work with the provided instructions above. Are you sure you exactly followed the described steps? Sadly i don't own a steamdeck for debugging. On the other hand the steamdeck uses an immutable desktop. Maybe this is the culprit.

Note you have to use `pacman -S` instead of `dnf install` to install the listed dependencies here.: 
https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1786192456

**Comment by Tnnk3** [comment]
Thanks, i did try using those but i am now getting errors: failed to init (unable to lock database)
error: could not lock database: read only file system
Was able to get past this and disabled readonly in steamos but the following error shows up now:
error: target not found: install
Warning: git 2-41.0-1 is up to date -- reinstalling
and all error after is the same as target not found for each dependencies

**Comment by fakhraldin** [comment]
I am probably not very smart to even think of assisting users from a remote place to compile a development driver on the steam deck. There is a reason, why that gaming machine has an immutable desktop and therefore read-only system files locked up from it's users. The intention of this project was actually to keep things as simple as possible for casual users at minimal risk and not to go deeper into their system and possibly break things.

So a disclaimer from my side: I can not help nor be held responsible in case people's SteamOS breaks. As mentioned above i don't own a steam deck, that seems to do some things quite differently than the distros arch and fedora. Debugging is limited and the help i can provide is to be noticed as experimental. So if you follow, you agree to the terms. 

Maybe i can come up with something. I have to figure out a few things first. In any case though backup your SteamOS. An alternative method would be to install arch in parallel to SteamOS maybe, to compile the driver on arch and then to try it out on SteamOS. But this depends again on the customized environment of SteamOS, how far away it is from the infrastructure of arch.

**Comment by Tnnk3** [comment]
Thanks! Appreciate you for trying to help. :)

**Comment by fakhraldin** [comment]
Ok, here we go. I hope you did a backup of your SteamOS in the mean time. First of all we have to unlock the partiiton on the steam deck. You already did so i know, but i write it down for others to follow up in case this succeeds.

1.) Make the partition writable: 
Set a password for the deck user:
`passwd`
Disable the readonly constraint:
`sudo steamos-readonly disable`

2.) Enable the pacman archlinux package manager:
Initialise the key database
`sudo pacman-key --init`
Populate the key database
`sudo pacman-key --populate`
Fix the pacman repository names as per https://www.reddit.com/r/SteamDeck/comments/zrc2ep/fail_to_run_any_pacman_commands/
(This is probably the reason why the install of the dependencies failed before.)
`sudo sed -i -E 's/^\[(holo)\]/[\1-rel]/' /etc/pacman.conf`
`sudo sed -i -E 's/^\[(core|jupiter|extra|community|multilib)\]/[\1-rel]/' /etc/pacman.conf`
update the pacman package database
`sudo pacman -Sy`
The steam deck base image does not come with /usr/include and other files required to build things, so we need to reinstall some of the packages, even though they already installed.
You should see a few warning messages about 'could not get file information /usr/share/...', because of that and you can safely ignore them. 

3.) Now try to install the dependencies again by starting from here.
https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1786192456

**Comment by Tnnk3** [comment]
Sadly, this doesn't work either. Its still saying target not found errors for all dependencies.

**Comment by andyktaylor** [comment]
> 3.) As a next stept we are going to configure the build with meson. Enter this command: `meson build64 --libdir lib64 --prefix $HOME/mesa -Dgallium-drivers=radeonsi,swrast,zink -Dvulkan-drivers=amd -Dgallium-nine=true -Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec -Dbuildtype=release`

Got this far on my Steamdeck running Bazzite, but when I paste in this command, I'm seeing:

Bash : meson: command not found

Any guidance please?

**Comment by andyktaylor** [comment]
Trying this on my Steamdeck, running Bazzite, but after running step 3, I get this - any idea on what I can do here to fix please?

[deck@fedora mesa]$ meson build64 --libdir lib64 --prefix $HOME/mesa -Dgallium-drivers=radeonsi,swrast,zink -Dvulkan-drivers=amd -Dgallium-nine=true -Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec -Dbuildtype=release
The Meson build system
Version: 1.2.3
Source dir: /var/home/deck/Projects/mesa
Build dir: /var/home/deck/Projects/mesa/build64
Build type: native build
Project name: mesa
Project version: 24.0.0-devel

meson.build:21:0: ERROR: Unable to detect linker for compiler `cc -Wl,--version`
stdout: 
stderr: collect2 version 13.2.1 20230728 (Red Hat 13.2.1-1)
[cannot find ld] -plugin /usr/libexec/gcc/x86_64-redhat-linux/13/liblto_plugin.so -plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/13/lto-wrapper -plugin-opt=-fresolution=/tmp/cc8YB6vs.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --no-add-needed --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 /usr/lib/gcc/x86_64-redhat-linux/13/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/13/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/13/crtbegin.o -L/usr/lib/gcc/x86_64-redhat-linux/13 -L/usr/lib/gcc/x86_64-redhat-linux/13/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/13/../../.. --version -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-redhat-linux/13/crtend.o /usr/lib/gcc/x86_64-redhat-linux/13/../../../../lib64/crtn.o
collect2: fatal error: cannot find ‘ld’
compilation terminated.

**Comment by fakhraldin** [comment]
> Sadly, this doesn't work either. Its still saying target not found errors for all dependencies.

I expected something like that. The repository renaming to `-rel` doesn't seem to match. I hoped `-rel` would serve us as an universal match in regard to repository names. So let's try to tailor this. What exact SteamOS version do you have? Replace `-rel` with your SteamOS version number, something like `-3.5.1` or higher according to your OS version.
```
sudo sed -i -E 's/^\[(holo)\]/[\1-3.5.1]/' /etc/pacman.conf
sudo sed -i -E 's/^\[(core|jupiter|extra|community|multilib)\]/[\1-3.5.1]/' /etc/pacman.conf
```

**Comment by andyktaylor** [comment]
> Sadly, this doesn't work either. Its still saying target not found errors for all dependencies.

FYI if you change to the latest beta, apparently the MESA drivers are already patched (i.e. Steam OS 3.5.1) and the game runs fine.

**Comment by fakhraldin** [comment]
> collect2: fatal error: cannot find ‘ld’
> compilation terminated.

I assume you already ruled out `ld` missing by error checking with `which ld` ? What does it say on Bazzite? 

Interesting Distro by the way. However i am focussing on SteamOS and arch on the steam deck first, because i want to know if the basic method can work there at all. This is to rule out possible main obstacles. 
Since you seem to be versed with installing a custom distro on the steam deck, would you mind to install arch on it on a separate storage medium and try out the mesa-build there? Otherwise we have to wait for a steam deck owner who can try that.

**Comment by fakhraldin** [comment]
> FYI if you change to the latest beta, apparently the MESA drivers are already patched (i.e. Steam OS 3.5.1) and the game runs fine.

Great News! That was fast :)

**Comment by samt1803** [comment]
> FYI if you change to the latest beta, apparently the MESA drivers are already patched (i.e. Steam OS 3.5.1) and the game runs fine.

Can someone confirm this? Or: how do I get that version? I was on beta candidate the issue was there, switched to Main, which I guess is the main branch, like a nightly, and it is also still there. It has still Mesa 23.3.0-devel (git-85a01d8fa0).

I by the way also tried building it on the deck with the info from here and some googling, but it resulted in many and endless builds not working because of missing dependencies, then trying to build those dependencies from AUR with yay, which is endless, just llvm-libs downloads gigabytes of data for the gitrepo..
The err that I got stuck on was "meson.build:901:6: ERROR: Could not get define 'ETIME'" on the meson step for mesa, but it would probably not be the last :)
Anyway was a fun bit of tinkering, thanks for looking into it! I think I'll wait for someone else to build it for/on the deck for now..

**Comment by andyktaylor** [comment]
Here's the guide to get it working, note the 3.5.1 requirement - turn on Developer options, then go into Developer settings and enable all beta channels, then back to Settings and select the PREVIEW channel

https://steamdeckhq.com/tips-and-guides/alan-wake-2-on-steam-deck/

**Comment by samt1803** [comment]
Ah let me clarify: I was on beta, version 3.5.1, game worked but still with the texture loading issue. Now I switched to 'main' which is even more bleeding edge / unstable and is version 3.6 and also the game works but still with the texture loading issue.
Mesa seems not to be patched and updated in these versions. 3.6 on main still has Mesa 23.3.0-devel (git-85a01d8fa0), which I don't know how old it is, but it was mentioned somewhere on Oct. 7th, so it is not very new, or at least not '0-day' new. Also official mesa beta (and also the base for this patched build here) is on 24 I think.

I also tried the built mesa versions from here, by fakhraldin and the one by Mikaka27, sadly both result in a black screen.

I also have it running on an external windows SSD, and there it runs fine, without texture issues, but with the occasional hard crash..

**Comment by mytrixxx** [comment]
I managed to compile for SteamOS 3.6 (main branch).
Tested, it works. Just unzip the mesa folder to /home/deck/ and add VK_ICD_FILENAMES=/home/deck/mesa/share/vulkan/icd.d/radeon_icd.x86_64.json %command%
https://gofile.io/d/3QDg0f

**Comment by samt1803** [comment]
> I managed to compile for SteamOS 3.6 (main branch). Tested, it works. Just unzip the mesa directory to /home/deck/ and add VK_ICD_FILENAMES=/home/deck/mesa/share/vulkan/icd.d/radeon_icd.x86_64.json %command% https://gofile.io/d/3QDg0f

It works! Amazing! Thanks a lot! 🙏 All of you!

**Comment by andyktaylor** [comment]
> I managed to compile for SteamOS 3.6 (main branch). Tested, it works. Just unzip the mesa directory to /home/deck/ and add VK_ICD_FILENAMES=/home/deck/mesa/share/vulkan/icd.d/radeon_icd.x86_64.json %command% https://gofile.io/d/3QDg0f

Perfect, working fine on my Bazzite install using this, nice one!

**Comment by mytrixxx** [comment]
One thing that doesn't work is the frame limiter.

**Comment by fakhraldin** [comment]
@mytrixxx  If that frame limiter is a custom overlay solution from valve, then it probably uses a tailored mesa pkg for SteamOS. Does mangohud work?

I am curious about the differences of SteamOS to a pc distro. Do you mind to elaborate what the needed steps were to make it finally work on the steam deck?

**Comment by mytrixxx** [comment]
Because I didn't want to break the OS on SteamDesk I installed it via Oracle VM VirtualBox.
Through sudo nano /etc/pacman.conf I changed all repositories to main. After the update, I tried to install all the packages according to the list. Then I followed your instructions. In the course of compilation I just added the missing python-mako, libgudev and systemd-libs packages. Then the compilation went without any problems.

Anyway, thank you for the compilation instructions.

**Comment by fakhraldin** [comment]
Nice! You are welcome! :)

**Comment by Tnnk3** [comment]
Thanks everyone! I got it working as well! Cheers!

**Comment by mytrixxx** [comment]
Gamescope limiter patch integrated,
https://gofile.io/d/BqRYc6

**Comment by laeitmotiv** [comment]
> Gamescope limiter patch integrated, https://gofile.io/d/BqRYc6

Thanks for the precious help! Unfortunately, when I try to run AW2 from Heroic on SteamOS 3.6 with the VK_ICD FILENAMES variable I get a black screen too. Could you please write a little bit more in depth the passages to compile myself the drivers?

**Comment by mytrixxx** [comment]
Weird. It works fine for me and other users on SteamOS 3.6.
Try the [original version without the gamescope limiter](https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1789417617).
Do you have all the paths set correctly? Does the game work from MESA with the default driver?
When compiling I used the guide above plus here are my [notes](https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1789569083) on it.

**Comment by laeitmotiv** [comment]
Yeah I tried the original version too before, no luck. :(

I'm pretty sure I'm doing something wrong, so before trying to compile it myself (since last time I compiled something by myself it was on Windows, years and years ago) I'll try to explain my whole situation here:

- I have a Steam Deck where I installed SteamOS 3.6 just this morning
- I downloaded AW2 from Heroic by linking my EGS account, so I have the newest update installed (1.0.8 iirc)
- I downloaded your .zip and extracted it on my /home/deck folder, so I have a /mesa folder in my Deck's home
- I added the environmental variable "VK_ICD_FILENAMES=/home/deck/mesa/share/vulkan/icd.d/radeon_icd.x86_64.json" in AW2's settings in Heroic
- I'm using the latest Wine-GE to run the game from Heroic
- Without trying to use your compiled drivers, the game correctly works, with the mesh shaders issue ofc

**Comment by andyktaylor** [comment]
If it makes any difference, I did mine in Lutris like @fakhraldin posted, rather than Heroic, but should work the same.

3.) Procedure:

Download the latest driver package from the release section of this project.

Decompress the zip file to your home directory: /home/(YOUR-USER-PROFILE-NAME)/

(Don't forget to replace (YOUR-USER-PROFILE-NAME) with your actual user name!)

Now you should see the following directory: /home/(YOUR-USER-PROFILE-NAME)/mesa

Open Lutris and right click on your game's icon. Choose "Config".

Go to the tab "system settings".

Scroll down to "environment variables" and click the "add" button right next to it.

In the field "key" enter: VK_ICD_FILENAMES

In the field "value" right next to it enter:

/home/(YOUR-USER-PROFILE-NAME)/mesa/share/vulkan/icd.d/radeon_icd.x86_64.json

Press the save button in the top right corner of the current config window.

**Comment by fakhraldin** [comment]
Good news and caution! 
When you patch and compile mesa-git on your machine, please take note that patch 25659 is not necessary anymore. The MR for it has been [merged](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25659) into mesa-git just 20 minutes ago. So only apply the other two patches left here: https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1786223949 if they make a difference for you.

The devs made some code changes to 25659, which might result in a progression or regression in performance and visual quality. So if you want to use the newest status of this mesh shaders code merge, you have to build mesa-git and refrain from using previously built ones including my releases. 
But they are still useful though as a reference to measure the difference between the old 25659 code in previous releases of this project and fresh mesa-git builds.

**Comment by Dante1475** [comment]
How can I patch and compile mesa-git on SteamOS? When I try to do the proccess of tutorial made for fakhraldin, the PC (Steam Deck) show me some errors like:

Unknow meson command
Missing pip comando
Dependencies error.

**Comment by samt1803** [comment]
@Dante1475 I wouldn't go down the compile route on the deck. My experience was that even after installing meson and python+pip there are many dependencies missing during the compile setup step with meson. And many of them are not available as binaries, meaning you need to also compile them from AUR, which again results in dependency issues. 

For me the versions compiled by @mytrixxx work perfectly, I recommend that: https://github.com/fakhraldin/mesa-git-radv-patched-25659/issues/1#issuecomment-1789893576

Apparently the relevant patch is also merged, so using a pre-compiled version of mesa-git might also work, like in the end of this video https://www.youtube.com/watch?v=QGMNH6Fqb1k but I don't know whether that flatpak right now already has the latest mesa-git branch, so not sure if it is in there, also I did not test this route..

**Comment by Dante1475** [comment]
> @Dante1475 
> 
> Apparently the relevant patch is also merged, so using a pre-compiled version of mesa-git might also work, like in the end of this video https://www.youtube.com/watch?v=QGMNH6Fqb1k but I don't know whether that flatpak right now already has the latest mesa-git branch, so not sure if it is in there, also I did not test this route..


I saw that video before, but after try again with he's method, Discover show an update of mesa-git, I install it and now the textures render well.
